### PR TITLE
Migrate check-ci step to buildkite

### PR DESCRIPTION
**Pull request by pazone**
## What does this PR do?
Migrates `check ci` step to buildkite

### CODE REVIEW

**Review by pchila — COMMENTED** [review]
Migration of the step looks ok, however jenkins pipeline is now broken
https://fleet-ci.elastic.co/blue/organizations/jenkins/elastic-agent%2Felastic-agent-mbp/detail/PR-3759/2/pipeline/157/

I suspect that removing 
```
setEnvVar('BEAT_VERSION', sh(label: 'Get beat version', script: 'make get-version', returnStdout: true)?.trim())
```
had some unforeseen side effects (or maybe there some other side effect of  `make check-ci` that is now missing :thinking:

**Comment by pazone** [comment]
@pchila We concluded that the package step is used by e2e tests that are deprecated. So we can remove it.

File: .buildkite/scripts/steps/check-ci.sh [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+set -euo pipefail
+
+source .buildkite/scripts/common.sh
+
+echo "--- Check CI"
+go version
+mage --version
+BEAT_VERSION=$(make get-version)
+echo "Beat version: $BEAT_VERSION"
```
Comment (alexsapran) [inline .buildkite/scripts/steps/check-ci.sh:11]: we could prefix this with `---` so this provides a better experience with BK logs.

File: .ci/Jenkinsfile [hunk 2]
```diff
@@ -65,90 +65,7 @@ pipeline {
...
-           dir("${BASE_DIR}"){
-              setEnvVar('BEAT_VERSION', sh(label: 'Get beat version', script: 'make get-version', returnStdout: true)?.trim())
-              log(level: 'INFO', text: "env.BEAT_VERSION=${env.BEAT_VERSION}")
-              cmd(label: 'check', script: 'make check-ci')
-            }
-          }
-        }
-      }
-    }
-    stage('Test') {
```
Comment (alexsapran) [inline .ci/Jenkinsfile:82]: While we migrate the check we remove additional `Test` and `Package` from Jenkins.

File: .buildkite/scripts/steps/check-ci.sh (hunk 1 above)
Comment (pazone) [reply .buildkite/scripts/steps/check-ci.sh:11]: The previous `echo "--- Check CI"` already has this prefix. Isn't it enough?

Comment (alexsapran) [reply .buildkite/scripts/steps/check-ci.sh:11]: I don't know, what is this `echo`?

**Comment by pazone** [comment]
@pchila @ycombinator Could you please revisit this PR?

**Review by pchila — APPROVED** [review]

**Pull request merged**